### ISSUE TITLE
[Bug]: Authorization not working: Claims are ignored

### ISSUE THREAD

**Issue by Ka-Narlist** [CoreWCF/CoreWCF#1408]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.5.2

### Describe expected behavior

I'm have a service method decorated with [Authorize(Policy = "Policy1")] and a policy with that name that expects the user to have a specific claim. the user doesn't have the claim and should get an error when calling the service method.

### Describe actual behavior

The user can call the method without error and the policy doesn't get evaluated at all.

### Which binding

BasicHttp

### security

Transport

### Which .NET version

.NET 6

### Which os platform

Windows

### Code snippet used to reproduce the issue

```c#
I have an example here https://github.com/Ka-Narlist/CoreWcfAuthorization
```

**Comment by Ka-Narlist** [comment]
One more thing. If I add this

`app.MapGet("/", () => "Test").RequireAuthorization("Policy1");`

and open the site in a browser I get the expected 403 error. If I change the not existing claim to an existing one, I get the "Test" output. So in principle this works, only CoreWCF doesn't seem to notice any of this.

**Comment by bjornbouetsmith** [comment]
I don't think claims has ever worked in a WCF context - so I think this should be a feature request and not a bug.
It would be nice to be able to annotate service methods with claims, but I doubt it has ever worked.

**Comment by g7ed6e** [comment]
This is actually supported. @Ka-Narlist Can you check you followed instructions from that blog post https://corewcf.github.io/blog/2023/02/19/aspnetcore-authorization-support

**Comment by Ka-Narlist** [comment]
> ```cs
> https://github.com/Ka-Narlist/CoreWcfAuthorization
> ```

Hey, thank you for your reply. I tried adding RequireAuthenticatedUser like this:
`options.AddPolicy("Policy1", policy =>
{
    policy.RequireAuthenticatedUser()
          .RequireClaim("http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid", "xyz");
});`

and like it's done in the JWTBearer example:
`options.AddPolicy("Policy1", new AuthorizationPolicyBuilder(NegotiateDefaults.AuthenticationScheme)
    .RequireAuthenticatedUser()
    .RequireClaim("http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid", "xyz")
    .Build());`

but either way it doesn't make a difference and the claim is not checked for calling the service method.

**Comment by Ka-Narlist** [comment]
> @Ka-Narlist can you share a minimal repo reproducing the issue you are facing ?

Yes, I updated the linked repo https://github.com/Ka-Narlist/CoreWcfAuthorization
